### DISCUSSION TITLE
changing veriants before loasding usdz

### DISCUSSION THREAD

**Discussion by MikePageDev** [google/model-viewer#4077] [Q&A]
Hi, I have a model that has some variants that can be changed by the user. this works perfectly when opening AR mode on an Android but not on ios. Is there a way to preset the variant on the usdz before it loads?

**Comment by elalish** [comment]
It appears to be working correctly on my iphone with this variants example: https://modelviewer.dev/examples/scenegraph/#variants

It should work automatically as long as you aren't specifying your own USDZ file. Is your site set up like the example? Perhaps you could post a screen recording?

**Reply by MikePageDev** [reply]
Hi @elalish 

Thank you for getting back to me. I got this work this morning. I missed the fact that you should not set the ios-src.
